src/implementation.js

```diff
@@ -0,0 +1,129 @@
+function _foldToObject(memo, [k, v]) {
+  memo[k] = v;
+  return memo;
+}
+
+function _objectMap(obj, fn) {
+  return _entries(obj).map(fn).reduce(_foldToObject, {});
+}
+
+function _entries(obj) {
+  let stringKeys = Object.getOwnPropertyNames(obj);
+  let symKeys = Object.getOwnPropertySymbols(obj);
+  return [...stringKeys, ...symKeys].map(p => [p, obj[p]]);
+}
+
+
+export default class Protocol extends null {
```

Comment:
Classes should not extend `null` as things stand. TC39 does not have consensus for what the semantics should be, and the current semantics mean that such classes cannot be instantiated. See https://github.com/tc39/ecma262/issues/1036, https://github.com/tc39/ecma262/pull/781, the [notes](https://github.com/rwaldron/tc39-notes/blob/master/es7/2017-01/jan-24.md#13ib-discussion-of-some-oddities-around-classes-extending-null), etc.

Reply:
Fixed: https://github.com/michaelficarra/proposal-first-class-protocols-polyfill/commit/7571e390805a092d13421a6c20252195a2b5791b